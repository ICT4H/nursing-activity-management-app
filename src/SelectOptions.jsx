import React, {Component} from 'react';
import PropTypes from "prop-types";

class SelectOptions extends Component {
  getAllOptions() {
    return this.props.options.map((o) => <option value={o} key={"option" + o}>{o}</option>);
  }

  render() {
    return (
        <select value={this.props.selectedValue} onChange={this.props.onChange}>
          <option>select Option</option>
          {
            this.getAllOptions()
          }
        </select>
    );
  }
}

SelectOptions.propTypes = {
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
};
export default SelectOptions
