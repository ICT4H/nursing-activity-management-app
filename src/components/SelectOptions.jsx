import React, {Component} from 'react';
import PropTypes from "prop-types";

class SelectOptions extends Component {
  getAllOptions() {
    return this.props.options.map((o) => <option value={o.name} key={"option" + o.name}>{o.name}</option>);
  }

  render() {
    return (
        <select value={this.props.selectedValue} onChange={this.props.onChange} className={this.props.className}>
          <option>{this.props.chooseMsg || ""}</option>
          {
            this.getAllOptions()
          }
        </select>
    );
  }
}

SelectOptions.propTypes = {
  options: PropTypes.arrayOf(PropTypes.object).isRequired,
};
export default SelectOptions
