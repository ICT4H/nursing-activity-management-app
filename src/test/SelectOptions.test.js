import React from 'react';
import SelectOptions from '../SelectOptions';
import renderer from 'react-test-renderer';

test('Should have all options from list of options provided', () => {
  let options=["male","female"];
  const component = renderer.create(
      <SelectOptions options={options} className="selectOption" chooseMsg="select Option"/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Given option should be selected default', () => {
  let options=["male","female"];
  let selectedValue="male";
  const component = renderer.create(
      <SelectOptions options={options} className="selectOption"
                     selectedValue={selectedValue} chooseMsg="select Option"/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});