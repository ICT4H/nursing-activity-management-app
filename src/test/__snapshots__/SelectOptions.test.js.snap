// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given option should be selected default 1`] = `
<select
  className="selectOption"
  onChange={undefined}
  value="male"
>
  <option>
    select Option
  </option>
  <option
    value="male"
  >
    male
  </option>
  <option
    value="female"
  >
    female
  </option>
</select>
`;

exports[`Should have all options from list of options provided 1`] = `
<select
  className="selectOption"
  onChange={undefined}
  value={undefined}
>
  <option>
    select Option
  </option>
  <option
    value="male"
  >
    male
  </option>
  <option
    value="female"
  >
    female
  </option>
</select>
`;
